core/src/main/java/com/valtech/aem/saas/core/bestbets/ClientBestBetsConsumerService.java

```diff
@@ -0,0 +1,189 @@
+        .build();
+    Optional<SearchResponse> searchResponse = searchRequestExecutorService.execute(searchRequest);
+    handleFailedRequestExecution(searchResponse);
+    searchResponse
+        .ifPresent(r -> handleSearchResponseError(r,
+            String.format("Failed to add best bets: %s", bestBetPayloadList)));
+  }
+
+  @Override
+  public void updateBestBet(int bestBetId, @NonNull BestBetPayload bestBetPayload) {
+    validateCommonConfigs();
+    if (StringUtils.isBlank(updateBestBetAction)) {
+      throw new IllegalStateException("Update best bet action path is not specified.");
+    }
+    SearchRequest searchRequest = SearchRequestPut.builder()
+        .uri(commonPath + updateBestBetAction + "/" + bestBetId)
+        .httpEntity(createJsonPayloadEntity(bestBetPayload))
+        .build();
+    Optional<SearchResponse> searchResponse = searchRequestExecutorService.execute(searchRequest);
+    handleFailedRequestExecution(searchResponse);
+    searchResponse.ifPresent(r -> handleSearchResponseError(r,
+        String.format("Failed to update best bet with id %s, with %s", bestBetId, bestBetPayload)));
+    if (!searchResponse
+        .flatMap(response -> response.get(new ModifiedBestBetIdExtractionStrategy()))
+        .isPresent()) {
+      throw new BestBetsActionFailedException(
+          String.format("Failed to update best bet: %s with the following update details %s", bestBetId,
+              bestBetPayload));
+    }
+  }
+
+  @Override
+  public void deleteBestBet(int bestBetId) {
+    validateCommonConfigs();
+    if (StringUtils.isBlank(deleteBestBetAction)) {
+      throw new IllegalStateException("Delete best bet action path is not specified.");
+    }
+    SearchRequest searchRequest = SearchRequestDelete.builder()
+        .uri(commonPath + deleteBestBetAction + "/" + bestBetId)
+        .build();
+    Optional<SearchResponse> searchResponse = searchRequestExecutorService.execute(searchRequest);
+    handleFailedRequestExecution(searchResponse);
+    searchResponse.ifPresent(r -> handleSearchResponseError(r,
+        String.format("Failed to delete best bet with id %s", bestBetId)));
+    if (!searchResponse
+        .filter(SearchResponse::isSuccess)
+        .flatMap(response -> response.get(new ModifiedBestBetIdExtractionStrategy()))
+        .isPresent()) {
+      throw new BestBetsActionFailedException(String.format("Failed to delete best bet: %s", bestBetId));
+    }
+  }
+
+  @Override
+  public void publishBestBetsForProject(int projectId) {
+    validateCommonConfigs();
+    if (StringUtils.isBlank(publishBestBetsForProjectAction)) {
+      throw new IllegalStateException("Publish best bets for project action path is not specified.");
+    }
+    if (!StringUtils.contains(publishBestBetsForProjectAction, "%s")) {
+      throw new IllegalArgumentException(
+          "Publish Best Bets For Project Action is of illegal format. It should contain a wildcard/placeholder for project id.");
```

Comment:
this should be checked earlier, e.g. during activation. if configs are not valid the compoennt shouldn't even start up